cpp/core/jni/JniWrapper.cc

```diff
@@ -913,15 +912,25 @@ JNIEXPORT void JNICALL Java_io_glutenproject_vectorized_OnHeapJniByteInputStream
   JNI_METHOD_END()
 }
 
-JNIEXPORT jlong JNICALL
-Java_io_glutenproject_vectorized_ShuffleReaderJniWrapper_make(JNIEnv* env, jclass, jobject jni_in, jlong c_schema) {
+JNIEXPORT jlong JNICALL Java_io_glutenproject_vectorized_ShuffleReaderJniWrapper_make(
+    JNIEnv* env,
+    jclass,
+    jobject jni_in,
+    jlong c_schema,
+    jlong alloc_id) {
   JNI_METHOD_START
+  auto* allocator = reinterpret_cast<MemoryAllocator*>(alloc_id);
+  if (allocator == nullptr) {
+    gluten::JniThrow("Memory pool does not exist or has been closed");
+  }
+  auto pool = AsWrappedArrowMemoryPool(allocator);
   std::shared_ptr<arrow::io::InputStream> in = std::make_shared<JavaInputStreamAdaptor>(env, jni_in);
   ReaderOptions options = ReaderOptions::Defaults();
+  options.ipc_read_options.memory_pool = pool.get();
```

Comment:
Is this option pool duplicated with Reader constructor pool?

Reply:
No, this would be passed to the underlying IPC reader.

The one in constructor is to keep the pool alive.